### PR TITLE
resolve chapter1

### PR DESCRIPTION
**Pull request by krivtsov**
### Solutions for Chapter 1

cc @vrom911 @chshersh

### CODE REVIEW

File: src/Chapter1.hs [hunk 1]
```diff
@@ -541,9 +542,11 @@ value after "=" where the condition is true.
...
 Casual reminder about adding top-level type signatures for all functions :)
 -}
-
-mid x y z = error "mid: not implemented!"
-
+mid :: Int -> Int -> Int -> Int
+mid x y z
+  | x >= y && x <= z || x < y && x > z = x
+  | y < x && y > z || y > x && y < z = y
+  | otherwise = z
```
Comment (chshersh) [inline src/Chapter1.hs:549]: Nice solution and correct usage of otherwise!

File: src/Chapter1.hs [hunk 2]
```diff
@@ -619,8 +629,11 @@ Implement a function that returns the sum of the last two digits of a number.
 Try to introduce variables in this task (either with let-in or where) to avoid
 specifying complex expressions.
 -}
-
-sumLast2 n = error "sumLast2: Not implemented!"
+sumLast2 :: Int -> Int
+sumLast2 n =
+  let secondNum = div (mod (abs n) 100) 10
+      lastNum = mod (abs n) 10
+  in secondNum + lastNum
```
Comment (chshersh) [inline src/Chapter1.hs:636]: That is a wonderful solution! 👏🏼  You correctly noticed that it is the `div` and `mod`, cool 😎 

One hint to make your solution even shorter: the standard library has the [`divMod`](https://hackage.haskell.org/package/base-4.14.0.0/docs/Prelude.html#v:divMod) function, that is actually combines inside both `div` and `mod`. And this is exactly what you use!.

So you could write it this way:

```haskell
(first, second) = divMod (mod x 100) 10
```

You can see how we could pattern match on the pair 🙂

**Review by chshersh — APPROVED** [review]
Nice work on solving our challenges!
And well done with the Advanced Task :medal_sports: 

I left my review and some comments. Hope you find them helpful :slightly_smiling_face:

**Comment by vrom911** [comment]
I'm going to close this PR, because we can't merge the solutions to our repo :slightly_smiling_face: 
Feel free to open PRs with solutions to other chapters!

**Pull request closed without merging**